apps/src/dance/Dance.js

```diff
@@ -485,16 +484,24 @@ Dance.prototype.preview = async function () {
 
   const charactersReferenced = utils.computeCharactersReferenced(studentCode);
   await this.nativeAPI.ensureSpritesAreLoaded(charactersReferenced);
-  this.hooks.find(v => v.name === 'runUserSetup').func();
 
-  // Force preview draw to occur **after** any
-  // draw iterations already queued up.
-  // redraw() (rather than draw()) is p5's recommended way
-  // of drawing once.
-  setTimeout(() => {
+  const previewDraw = () => {
+    this.nativeAPI.setEffectsInPreviewMode(true);
+
+    // the user setup hook initializes effects,
+    // needs to happen in preview mode for some effects (eg, tacos)
+    this.hooks.find(v => v.name === 'runUserSetup').func();
+
+    // redraw() (rather than draw()) is p5's recommended way
+    // of drawing once.
     this.nativeAPI.p5_.redraw();
-    this.nativeAPI.setForegroundEffectsInPreviewMode(false);
-  }, 0);
+
+    this.nativeAPI.setEffectsInPreviewMode(false);
+  };
+
+  // This is the mechanism p5 uses to queue draws,
+  // so we do the same so we end up after any queued draws.
```

Comment:
Nit: could you explain what you mean by 'so we do the same so we end up after any queued draws.'?

Reply:
Would it be clearer if it was "so we use the same mechanism" instead of "we do the same"? I can update next time I'm in here

Reply:
Oh -gotcha! I think maybe it would be more clear if you said '...so we use the same mechanism to ensure that this preview is drawn after any queued draw calls'.